src/GraphQL.SystemTextJson/ObjectDictionaryConverter.cs

```diff
@@ -11,6 +9,7 @@ namespace GraphQL.SystemTextJson
     /// A custom JsonConverter for reading a dictionary of objects of their real underlying type.
     /// Doesn't support write.
     /// </summary>
+    [Obsolete("This class will be removed in a future version of GraphQL.NET. Please use the InputsConverter instead.")]
```

Comment:
What to do those who use the properties of the `Dictionary<string, object` type? Do they have to change their type to `Inputs`? It will look strange.

Reply:
I thought we only have `Dictionary<string, object>` support to facilitate reading variables.  If this is not the case, we can discard this PR.  That's why it's a separate PR from my prior changes.

Reply:
It just seems to me that the "reading" helper methods of `GraphQL.SystemTextJson` are only there to help reading typical json graphql requests.  These requests typically have these properties:
- Query (as string)
- OperationName (as string)
- Variables (as Inputs object)

Any other data to be deserialized is outside the scope of the `GraphQL.SystemTextJson` library.  But most other properties can be handled natively by `System.Text.Json`.  For example any of this stuff can be handled by simply adding more properties to the class being deserialized:

- Username & password (string fields)
- JWT token (string field)
- API handle (guid)

You don't need an `ObjectDictionaryConverter` for that.  But again, support for these arbitrary and unknown properties are outside the scope of the `GraphQL.SystemTextJson` library, it would seem to me.

I suggest making the changes suggested in this PR, and if we get feedback that these methods are necessary, we can keep the methods and classes within v5.  I also suggest we make the breaking change (breaking for the last 2 days only so far) to remove the `ObjectDictionaryConverter` support from the new `FromJson` methods, as it would be improper to mark the entire method as obsolete.

Reply:
I have some concerns... Look at `GraphQL.Server.Transports.AspNetCore.SystemTextJson.InternalGraphQLRequest.Extensions` property of type `Dictionary<string, object>`. `GraphQL.Server.Transports.AspNetCore.SystemTextJson.GraphQLRequestDeserializer` uses `ObjectDictionaryConverter`. The siblings classes from `GraphQL.Server.Transports.AspNetCore.NewtonsoftJson` package do not use `ObjectDictionaryConverter` because I think conversion for dictionaries work out of the box there. So I think problems may arise (if any) only for STJ.

Reply:
While that makes some sense, it doesn't apply to the Server project.  The transport returns a `GraphQLRequest` object, and within that class, both `Variables` and `Extensions` are of type `Inputs`.  The `Inputs` class implements `IReadOnlyDictionary<string, object>` for reading data.  The `InternalGraphQLRequest` type is `internal` and both the `Variables` and `Extensions` properties should be changed to be of type `Inputs`, simplifying the code.  The deserializer should utilize the `InputsConverter` instead of the `ObjectDictionaryConverter`.  There will be no functionality difference and no public API signature changes.  If anything, it would seem to indicate that this PR should be merged.

Reply:
See https://github.com/graphql-dotnet/server/pull/569

Reply:
@sungam3r Do you have further objections after seeing how this can be implemented - as demonstrated by the changes in the server project?

Keep in mind, we are just marking as obsolete, and can remove the obsolete attribute if needed before v5 releases.

Reply:
No objections.